### PR TITLE
Contribution guidelines for the SunPy Example Gallery

### PR DESCRIPTION
**Pull request by mbobra**
@drsophiemurray and I developed some contribution guidelines for the example gallery and integrated these into the developer guide. Please let us know what you think!

### CODE REVIEW

File: docs/dev_guide/example_gallery.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+The purpose of the page is to describe the contribution guidelines for the `SunPy Example Gallery <http://docs.sunpy.org/en/stable/generated/gallery/index.html>`_. All potential contributors to the SunPy Example Gallery should read and abide by the following guidelines.
+
+Contribution Guidelines
+-----------------------
+
+-  The title of the example should be short yet descriptive and emphasize the goal of the example. Try to make the title appeal to a broad audience and avoid referencing a specific instrument, catalog, or anything wavelength dependent.
+-  Each example should begin with a paragraph that gives a brief overview of the entire example, including relevant astronomy concepts, and motivates the described functionality.
+-  The examples must be compatible with the versions supported by the last major release of the SunPy core package (i.e. Python >= 3.6).
+-  All the examples should be checked using one of the many PEP8 linters that are available (autopep8, pylint, flake8 as some examples).
```
Comment (Cadair) [inline docs/dev_guide/example_gallery.rst]: I would re-write this one to say something like, all the examples must be strictly PEP8 compliant. There are tools available to check this for you, i.e. flake8 ...

File: docs/dev_guide/example_gallery.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+-  Wherever possible, the examples should include linked references with links pointing to the appropriate ADS entry.
```
Comment (Cadair) [inline docs/dev_guide/example_gallery.rst]: ADS entry or DOI ? just incase ADS missed something?

File: docs/dev_guide/example_gallery.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+-  The example should include links to relevant documentation pages.
+-  Each example should include at least one image, map, or plot to use as the icon in the Example Gallery.
```
Comment (Cadair) [inline docs/dev_guide/example_gallery.rst]: I dont think this should be a hard requirement, there are somethings that are worthy of examples but dont involve plotting. I suggest we reword it to "if possible" or something.

Comment (mbobra) [reply docs/dev_guide/example_gallery.rst]: sounds good @Cadair -- I incorporated all these suggestions!

**Comment by ehsteve** [comment]
Thanks @mbobra! We definitely needed this.

**Comment by Cadair** [comment]
I hope you don't mind @mbobra but I tided this up so we can get it merged :smiley:

**Comment by mbobra** [comment]
Not at all @Cadair -- thanks for your help! 👍
